### DISCUSSION TITLE
Transpose a case from PeleLMeX to PeleC.

### DISCUSSION THREAD

**Discussion by lucasbrah** [AMReX-Combustion/PeleC#888] [Q&A]
Hi,
I'd like to transfer a case from PeleLMeX to PeleC. The PeleLMeX case uses EB (globally a jet trough a crossflow) but I don't think there is anything specific to do with PeleC. I was just wondering how to transpose mu PeleLMeX case to PeleC, whether it could be done quickly or not? Thank you very much for your suggestions!

**Comment by baperry2** [comment]
We support this in the other direction (going from C to LMeX). Most of the capability is there for go from LMeX to C, but there is a bit you'll have to add. In PeleC, use the restart from plot file option (`pelec.init_pltfile = "plt00000"`), and point that to your LMeX plot file. This will take rho, species (massfractions not rhoY), and temperature from the plot file and use them to initialize the state in PeleC, so all those quantities must be in the plot file. 

The catch is some variables have different names and different units in PeleC vs. PeleLMeX, so you'd have to modify the routines that read from the plot file to account for this. Here's the [PeleC function](https://github.com/AMReX-Combustion/PeleC/blob/development/Source/IO.cpp#L747) that would need to be modified, and here's the existing [PeleLMeX function](https://github.com/AMReX-Combustion/PeleLMeX/blob/652f0eaeadef711653757f66de83ce4a124fd4c9/Source/PeleLMeX_Plot.cpp#L819) that can go in the opposite direction.

If you implement this, please consider submitting a PR with the capability.

**Reply by lucasbrah** [reply]
Thanks for your help, I'm generally new to the code. The fact of initializing by a plotfile of PeleLMeX on PeleC will allow me to have a field already established. However my question was more about getting the prob.cpp, prob.H files, in order to get the executable, I'll take inspiration from the pelelmex_prob.cpp files, etc. from my case. Is this a viable and simple approach?

Thank you !

**Reply by baperry2** [reply]
Yes, adapting the `pelelmex_prob.cpp` to `prob.cpp`, etc, should be a viable strategy. For simple cases this should be straightforward, but if you're doing more complex stuff beyond the bcnormal and initdata, it can get a bit tricky. 

One thing to keep in mind is that PeleC uses cgs units while PeleLMeX uses mks units. Another is that the bcnormal function for PeleLMeX is interpreted as applying values on the *boundary faces*, while in PeleC it populates data in *ghost cell centers*.
